### ISSUE TITLE
Make background drawing faster, most notably image decoding

### ISSUE THREAD

**Issue by jwijenbergh** [qtile/qtile#4826]
### Issue description

Currently background drawing is very slow (obviously depends on the image). The main culprit is the decoding of the wallpaper using `cairocffi.pixbuf.decode_to_image_surface`. This is really problematic as we have to re-render the wallpaper every time the screen layout changes. On wayland this can even result in short freezes when you disable/enable monitors. Possible approaches:

- cache the image data, maybe not smart if large
- do some sort of async thing with decoding?

ideas?

### Version

git

### Backend

Wayland (experimental)

### Required

- [X] I have searched past issues to see if this bug has already been reported, and it hasn't been.
- [X] I understand that people give their precious time for free, and thus I've done my very best to make this problem as easy as possible to investigate.

**Comment by tych0** [comment]
caching it seems reasonable, how big is one of these when they're parsed?

but i wonder: can we just not re-paint the background if it's the same after the screens are reconfigured?

**Comment by jwijenbergh** [comment]
> how big is one of these when they're parsed

`sys.getsizeof(image)` only returns 56 bytes haha, not sure if that is completely correct

> but i wonder: can we just not re-paint the background if it's the same after the screens are reconfigured?

Yeah that is a good point. I could look at this, now we generate a whole new surface/buffer

**Comment by elParaguayo** [comment]
@tych0 I don't think this is urgent for the next release. If wayland is now working on wlroots 0.17, we should do a release now.

**Comment by jwijenbergh** [comment]
Yeah it's not, this has always been here

**Comment by tych0** [comment]
> If wayland is now working on wlroots 0.17, we should do a release now.

ok! I will go ahead and start the process :)

**Comment by jwijenbergh** [comment]
Merci! I merged some last minute fixes, so good to go from my end

**Comment by tych0** [comment]
https://github.com/qtile/qtile/releases/tag/v0.26.0 everything is grinding away, I will keep an eye on it and make sure it all goes green.

**Comment by jwijenbergh** [comment]
Green! Woohoo! Thanks

**Comment by jwijenbergh** [comment]
@github-actions I eat you for breakfast

**Comment by jwijenbergh** [comment]
@github-actions wanna take it outside
